### PR TITLE
[Static Runtime] Name refactoring to make MemoryPlanning more readable

### PR DESCRIPTION
**Pull request by penguinwu**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54272 More name refactoring of memory planning codes to make it more readable
* **#54045 [Static Runtime] Name refactoring to make MemoryPlanning more readable**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27233880](https://our.internmc.facebook.com/intern/diff/D27233880)

### CODE REVIEW

File: torch/csrc/jit/runtime/static/impl.h [hunk 1]
```diff
@@ -76,13 +77,25 @@ class TORCH_API StaticModule {
...
       std::pair<
           std::shared_ptr<torch::jit::Graph>,
           c10::optional<c10::FunctionSchema>> graph_and_schema,
       const StaticModuleOptions& opts);
 
+  // for <kind, idx>
+  //   if kind == CONSTANT_KIND: map to constants_[idx]
+  //   if kind == INPUT_KIND: map to inputs_[idx]
+  //   otherwise: map to nodes_[kind].outputs()[idx]
+  typedef std::pair<VALUE_KIND, int> DEF_INFO;
```
Comment (bertmaher) [inline torch/csrc/jit/runtime/static/impl.h:97]: A minor naming convention thing, usually our typedefs are camel case, like `DefInfo`

Comment (penguinwu) [reply torch/csrc/jit/runtime/static/impl.h]: Good to know. Updated to `DefInfo`.

File: torch/csrc/jit/runtime/static/impl.cpp [hunk 2]
```diff
@@ -251,7 +251,7 @@ LivenessInformation GetLivenessInformation(
 }
 
 // Implementation specific pruning of values
-// from "optimzable" set.  GetLivenessInformation and FindShared
+// from "optimzable" set.  GetLivenessInformation and FindSameStorageValues
```
Comment (bwasti) [inline torch/csrc/jit/runtime/static/impl.cpp:254]: Might even make sense to call this "GenerateSameStorageValues" as all the decision making happens here

File: torch/csrc/jit/runtime/static/impl.h [hunk 3]
```diff
@@ -76,13 +78,25 @@ class TORCH_API StaticModule {
       const torch::jit::Module& m,
       const StaticModuleOptions& opts = StaticModuleOptions());
 
+  typedef enum {
```
Comment (bwasti) [inline torch/csrc/jit/runtime/static/impl.h:81]: awesome

File: torch/csrc/jit/runtime/static/impl.h [hunk 4]
```diff
@@ -76,13 +78,25 @@ class TORCH_API StaticModule {
...
+    CONSTANT_VALUE = -2, // VALUE nodes defined by prim::Constant
+    INPUT_VALUE = -1, // VALUE nodes representing graph inputs
+    OTHER_VALUE = 0 // other VALUE nodes (use non-negative index)
```
Comment (bwasti) [inline torch/csrc/jit/runtime/static/impl.h:84]: it might be okay to omit `OTHER_VALUE`

File: torch/csrc/jit/runtime/static/impl.h [hunk 5]
```diff
@@ -16,6 +16,8 @@ struct TORCH_API StaticModuleOptions {
   bool cleanup_activations{true};
   bool enable_out_variant{true};
   bool optimize_memory{true};
+  bool optimize_output_memory{
+      false}; // to enable MemoryPlanner on output tensors
```
Comment (bwasti) [inline torch/csrc/jit/runtime/static/impl.h:20]: maybe comment above this line to avoid the line break?

**Review by bwasti — APPROVED** [review]
looks good! nice cleanup

File: torch/csrc/jit/runtime/static/impl.cpp (hunk 2 above)
Comment (penguinwu) [reply torch/csrc/jit/runtime/static/impl.cpp:259]: Done

File: torch/csrc/jit/runtime/static/impl.h (hunk 5 above)
Comment (penguinwu) [reply torch/csrc/jit/runtime/static/impl.h:20]: Fixed

File: torch/csrc/jit/runtime/static/impl.h (hunk 4 above)
Comment (penguinwu) [reply torch/csrc/jit/runtime/static/impl.h:84]: removed

**Pull request closed without merging**